src/Aspire.Hosting/Publishing/ManifestPublishingContext.cs

```diff
@@ -243,19 +243,21 @@ public void WriteServiceDiscoveryEnvironmentVariables(IResource resource)
                     ? endpointReferenceAnnotation.Resource.Annotations.OfType<EndpointAnnotation>().Select(sb => sb.Name)
                     : endpointReferenceAnnotation.EndpointNames;
 
-                var endpointAnnotationsGroupedByScheme = endpointReferenceAnnotation.Resource.Annotations
```

Comment:
Can we delete all of this code and rely on the `CreateEndpointReferenceEnvironmentPopulationCallback`? Now that we're setting EndpointReference directly, that should handle the formatting.